app/src/main/java/de/danoeh/antennapod/fragment/ItemlistFragment.java

```diff
@@ -638,14 +646,15 @@ private void loadItems() {
                 });
     }
 
-    private Pair<Feed, LongList> loadData() {
+    private Object[] loadData() {
         Feed feed = DBReader.getFeed(feedID);
         if(feed != null && feed.getItemFilter() != null) {
             FeedItemFilter filter = feed.getItemFilter();
             feed.setItems(filter.filter(feed.getItems()));
         }
         LongList queuedItemsIds = DBReader.getQueueIDList();
-        return Pair.create(feed, queuedItemsIds);
+        LongList favoritedItemsId = DBReader.getFavoriteIDList();
```

Comment:
Why do we need to query the database for the list of all favorites?  Isn't it enough to check if the item has the favorite tag or not?  For Queue we sometimes need the list because we need to know if something is at the top or bottom of the queue, but I don't think we need that context here.